Add requirements for setup to Notifier "k8sbundle" document

ryysud opened the pull request:
Signed-off-by: Ryuma Yoshida <ryumyosh@zlab.co.jp>

<!--
1. If this is your first PR, please read our contributor guidelines
https://github.com/spiffe/spiffe/blob/master/CONTRIBUTING.md
https://github.com/spiffe/spire/blob/master/CONTRIBUTING.md
-->

**Pull Request check list**

- [x] Commit conforms to CONTRIBUTING.md?
- [ ] Proper tests/regressions included?
- [x] Documentation updated?

**Affected functionality**
<!-- Please provide a description of the affected functionality -->

Nothing. I updated only documentation.

**Description of change**
<!-- Please provide a description of the change -->

When I tried to enable [Notifier "k8sbundle"](https://github.com/spiffe/spire/blob/v0.9.3/doc/plugin_server_notifier_k8sbundle.md), the requirements for setup it were not documented, so I added it. If the action added in this PR is not performed, the plugin prints an error and fails to run SPIRE server.

FYI, The steps to enable Notifier "k8sbundle" is as below:

Create a Kubernetes cluster.

```shell
$ cat config.yaml
kind: Cluster
apiVersion: kind.x-k8s.io/v1alpha4
kubeadmConfigPatches:
  - |
    kind: ClusterConfiguration
    metadata:
      name: config
    apiServer:
      extraArgs:
        "service-account-key-file": "/etc/kubernetes/pki/sa.pub"
        "service-account-signing-key-file": "/etc/kubernetes/pki/sa.key"
        "service-account-issuer": "api"
        "service-account-api-audiences": "api,spire-server"
  - |
    kind: KubeletConfiguration
    readOnlyPort: 10255
nodes:
  - role: control-plane
  - role: worker
  - role: worker

$ kind create cluster --name spire-dev --image kindest/node:v1.18.0 --config config.yaml
```

Deploy SPIRE Server v0.9.3 and SPIRE Agent v0.9.3 with Notifier "k8sbundle" enabled.

```shell
$ git clone https://github.com/spiffe/spire-examples.git
$ cd spire-examples
$ git rev-parse --short HEAD
2ef05ba

$ cat enable_k8sbundle.patch
diff --git a/examples/k8s/simple_psat/spire-agent.yaml b/examples/k8s/simple_psat/spire-agent.yaml
index 5aac447..f50a3e7 100644
--- a/examples/k8s/simple_psat/spire-agent.yaml
+++ b/examples/k8s/simple_psat/spire-agent.yaml
@@ -47,7 +47,7 @@ data:
       server_address = "spire-server"
       server_port = "8081"
       socket_path = "/run/spire/sockets/agent.sock"
-      trust_bundle_path = "/run/spire/config/bootstrap.crt"
+      trust_bundle_path = "/run/spire/bundle/bundle.crt"
       trust_domain = "example.org"
     }

@@ -74,19 +74,6 @@ data:
           }
       }
     }
-  bootstrap.crt: |
-    -----BEGIN CERTIFICATE-----
-    MIIBzDCCAVOgAwIBAgIJAJM4DhRH0vmuMAoGCCqGSM49BAMEMB4xCzAJBgNVBAYT
-    AlVTMQ8wDQYDVQQKDAZTUElGRkUwHhcNMTgwNTEzMTkzMzQ3WhcNMjMwNTEyMTkz
-    MzQ3WjAeMQswCQYDVQQGEwJVUzEPMA0GA1UECgwGU1BJRkZFMHYwEAYHKoZIzj0C
-    AQYFK4EEACIDYgAEWjB+nSGSxIYiznb84xu5WGDZj80nL7W1c3zf48Why0ma7Y7m
-    CBKzfQkrgDguI4j0Z+0/tDH/r8gtOtLLrIpuMwWHoe4vbVBFte1vj6Xt6WeE8lXw
-    cCvLs/mcmvPqVK9jo10wWzAdBgNVHQ4EFgQUh6XzV6LwNazA+GTEVOdu07o5yOgw
-    DwYDVR0TAQH/BAUwAwEB/zAOBgNVHQ8BAf8EBAMCAQYwGQYDVR0RBBIwEIYOc3Bp
-    ZmZlOi8vbG9jYWwwCgYIKoZIzj0EAwQDZwAwZAIwE4Me13qMC9i6Fkx0h26y09QZ
-    IbuRqA9puLg9AeeAAyo5tBzRl1YL0KNEp02VKSYJAjBdeJvqjJ9wW55OGj1JQwDF
-    D7kWeEB6oMlwPbI/5hEY3azJi16I0uN1JSYTSWGSqWc=
-    -----END CERTIFICATE-----

 ---

@@ -120,12 +107,15 @@ spec:
           args: ["-t", "30", "spire-server:8081"]
       containers:
         - name: spire-agent
-          image: gcr.io/spiffe-io/spire-agent:0.9.0
+          image: gcr.io/spiffe-io/spire-agent:0.9.3
           args: ["-config", "/run/spire/config/agent.conf"]
           volumeMounts:
             - name: spire-config
               mountPath: /run/spire/config
               readOnly: true
+            - name: spire-bundle
+              mountPath: /run/spire/bundle
+              readOnly: true
             - name: spire-agent-socket
               mountPath: /run/spire/sockets
               readOnly: false
@@ -145,6 +135,9 @@ spec:
         - name: spire-config
           configMap:
             name: spire-agent
+        - name: spire-bundle
+          configMap:
+            name: spire-bundle
         - name: spire-agent-socket
           hostPath:
             path: /run/spire/sockets
diff --git a/examples/k8s/simple_psat/spire-server.yaml b/examples/k8s/simple_psat/spire-server.yaml
index e921cd4..2e3bf04 100644
--- a/examples/k8s/simple_psat/spire-server.yaml
+++ b/examples/k8s/simple_psat/spire-server.yaml
@@ -25,6 +25,10 @@ rules:
 - apiGroups: ["authentication.k8s.io"]
   resources: ["tokenreviews"]
   verbs: ["create"]
+# Required for Notifier "k8sbundle" to push new trust bundle to configmap
+- apiGroups: [""]
+  resources: ["configmaps"]
+  verbs: ["get", "patch"]

 ---

@@ -112,6 +116,12 @@ data:
           cert_file_path = "/run/spire/config/bootstrap.crt"
         }
       }
+
+      Notifier "k8sbundle" {
+        plugin_data {
+        }
+      }
+
     }
   bootstrap.crt: |
     -----BEGIN CERTIFICATE-----
@@ -129,6 +139,14 @@ data:

 ---

+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: spire-bundle
+  namespace: spire
+
+---
+
 apiVersion: apps/v1
 kind: StatefulSet
 metadata:
@@ -151,7 +169,7 @@ spec:
       serviceAccountName: spire-server
       containers:
         - name: spire-server
-          image: gcr.io/spiffe-io/spire-server:0.9.0
+          image: gcr.io/spiffe-io/spire-server:0.9.3
           args: ["-config", "/run/spire/config/server.conf"]
           ports:
             - containerPort: 8081

$ git apply enable_k8sbundle.patch

$ kubectl apply -f examples/k8s/simple_psat/spire-server.yaml
$ kubectl apply -f examples/k8s/simple_psat/spire-agent.yaml
```

SPIRE Server and SPIRE Agent are running.

```shell
kubectl get all -n spire
NAME                    READY   STATUS    RESTARTS   AGE
pod/spire-agent-jhmc7   1/1     Running   0          56s
pod/spire-agent-q8qtv   1/1     Running   0          56s
pod/spire-server-0      1/1     Running   0          65s

NAME                   TYPE       CLUSTER-IP     EXTERNAL-IP   PORT(S)          AGE
service/spire-server   NodePort   10.96.171.80   <none>        8081:32040/TCP   65s

NAME                         DESIRED   CURRENT   READY   UP-TO-DATE   AVAILABLE   NODE SELECTOR   AGE
daemonset.apps/spire-agent   2         2         2       2            2           <none>          56s

NAME                            READY   AGE
statefulset.apps/spire-server   1/1     65s
```

**Which issue this PR fixes**
<!-- optional. `fixes #<issue number>` format will close an issue when this PR is merged -->

No issue.

evan2645 commented:
@ryysud also, need to catch this branch up with master - same permissions problem as last time.

Thanks again for this contrib!

ryysud commented:
@evan2645 
I addressed comment with https://github.com/spiffe/spire/pull/1490/commits/e4ea39da437b2854371ab0b4d70d80effa5229d9 and made the master branch up-to-date!

azdagron commented:
Thanks @ryysud!
